modules/fluid_properties/src/userobjects/FluidPropertiesInterrogator.C

```diff
@@ -131,7 +131,7 @@ FluidPropertiesInterrogator::execute()
 
       Moose::out << "**START JSON DATA**\n";
       Moose::out << json << "\n";
-      Moose::out << "**END JSON DATA**\n";
+      Moose::out << "**END JSON DATA**" << std::endl;
```

Comment:
Somebody really loves `std::endl` :-). Do we really need all those flushes?

Reply:
The flushes are required to make sure that the perf output doesn't clobber AFAIK

Reply:
Curious, I'd have thought output through the same stream always retains its order. I thought the rule is to flush as little as possible. Some exceptions are flushing right before user input is expected, or after a task is completed and a period of inactivity is expected (to make sure log files are up to date).

Reply:
If the flush is not in something like an element loop, we do not need to care imo.

Reply:
@permcody knows the reason, but I do believe we want to minimizing the flushing. All I can remember is my dad telling me not to waste water...if it is yellow let it mellow, if it is brown flush it down

Reply:
The same stream always retains its order ... with output mixed between two streams that eventually get interleaved, though, like the typical stdout+stderr pairing, less frequent flushing changes the ordering of results.  Usually you want stderr to be flushed as frequently as is sensible since that's where you put your "something's going down you need to know right now" messages, whereas with stdout you can heavily buffer most of its "work proceeding as planned, you can check it out when it's all done" messages.

Reply:
Oh, but about the interleaving - often you want to flush stdout *because of* (and immedately prior to) an error message to stderr, so that the stderr message has all of its context displayed, even if that context wasn't originally expected to be urgent when it got stuffed in the stdout buffer.

Reply:
Firstly: @dschwen ... `std::endl` and `std::flush` were only added at the very end of long prints - never in tight loops.  There _should_ have been flushes there anyway... for one thing, we don't want the OS buffering stuff forever and users not getting any output.  Too many endls or flushes is certainly not the objective though: they should be used when they should be used.

@roystgnr : The stream does not always retain its order when doing _threaded_ output.  In this PR there is now a _thread_ that is also printing to std:cout... so you can literally get garbarge output.  I've used locks to make sure that actual garbage doesn't come out - but we still don't want normal blocks of text being printed to get PerfGraph stuff interleaved into it too.

To alleviate this - all objects are writing to their own stringstreams... and the contents of that stringstream are only sent to the console once a std::endl or std::flush is encountered.  In this way we can actually guarantee that an entire block of text being written at one time by an object will come out on the console as a contiguous block of text with NO PerfGraph stuff in the middle of it.

BTW: The flushes were always happening (even in the current design)... they were just being done inside ConsoleOutput.